src/libos/src/signal/do_sigreturn.rs

```diff
@@ -224,8 +225,21 @@ fn handle_signals_by_user(
             const BIG_ENOUGH_SIZE: u64 = 4096;
             BIG_ENOUGH_SIZE as usize
         };
-        // TODO: validate the memory range of the stack
-        unsafe { Stack::new(stack_top, stack_size)? }
+
+        let stack = unsafe { Stack::new(stack_top, stack_size)? };
```

Comment:
The code works, but should be adjusted for better readability: the range of the stack should be validated first, then create a `Stack`. 

```rust
unsafe { Stack::new(stack_top, stack_size)? };
```

This is because the `unsafe` code block above essentially says: I am totally aware that creating a new stack object from the memory range `stack_top - stack_size` to `stack_top` is risky, but I do it anyway because I am certain that the range is safe to use. Thus, validating the range after creating the stack object is illogical.